src/app/treasure-hunt/treasure-hunt-report/treasure-hunt-report.component.ts

```diff
@@ -166,4 +176,43 @@ export class TreasureHuntReportComponent implements OnInit {
     this.printOptionsMenuService.showPrintMenu.next(true);
   }
 
+  showExportModal() {
+    this.fileName = this.getFileName();
+    this.exportModal.show();
+  }
+
+  hideExportModal() {
+    this.exportModal.hide();
+  }
+
+  getCurrentDate(): string{
+    const date: Date = new Date();
+    let formatedDate: string = moment(date).format("MMM D, YYYY").toString();
+    return formatedDate;
+  }
+
+  getFileName(): string {
+    if (!this.fileName) {
+      this.fileName = this.getCurrentDate() + ' - Treasure Hunt Report';
+    }
+    return this.fileName;
+  }
+
+  present() {
```

Comment:
The local variable facilityName reads a little confusing. It's the title 'Treasure Hunt Report", then it's the facility name + the title if we have a facility name. Then taken as the argument named facilityName (but it's the main slide title?) in the method.

You can remove the facilityName param and perform the logic from lines 207/208 inside the createPPT method. Settings is already being passed